CODE_GENERATION.md

```diff
@@ -350,25 +350,46 @@ go run cmd/provider/main.go
 
 ##### In-cluster
 
-You can build the images and deploy it into cluster using Crossplane Package Manager
+You can build the images and deploy it into the cluster using Crossplane Package Manager
 to test the real-world scenario. There are two Docker images to be built; one has
 the controller image, the other one has package metadata and CRDs. We need to choose a
 controller image before building metadata image. Then we'll start the build process.
 
-Example image URL pair:
-* `username/provider-aws-controller:test`: Controller Image
-  This is the one that will be running in the cluster.
-* `username/provider-aws:test`: Metadata Image
-  This is the one we'll use when we install the provider.
+Pre-requisite: 
+* Install Crossplane on the kind cluster following the steps [here](https://crossplane.io/docs/v1.1/reference/install.html).
+* Install Crossplane CLI following the steps [here](https://crossplane.io/docs/v1.1/getting-started/install-configure.html#install-crossplane-cli). 
+* Docker hub account with your own public repository.
 
-1. Edit `package/crossplane.yaml` and change the `spec.controller.image` with your controller image URL.
-1. Go to `package` folder and build the package using Crossplane CLI: `kubectl crossplane build provider`.
-1. Push the images using: `kubectl crossplane push provider <metadata image URL>`.
+Follow the steps below to get set-up for in-cluster testing:
+* Run `make build` in `provider-aws`. This will create two images mentioned above:
```

Comment:
If you run `make build DOCKER_REGISTRY=username VERSION=test-version`, then you can use the images directly without needing the `kubectl crossplane build provider` step because the `spec.controller.image` will be [correct](https://github.com/crossplane/provider-aws/blob/master/cluster/images/provider-aws/Makefile#L22) in this image. Though you still need to tag the final images because `VERSION` given here does not affect the final image tag.

Reply:
@muvaf I am not sure what this means - `Though you still need to tag the final images because VERSION given here does not affect the final image tag.`?
What would change in the subsequent steps if we run this - `make build DOCKER_REGISTRY=username VERSION=test-version`?

Reply:
If you run `make build DOCKER_REGISTRY=username VERSION=test-version`, you will still get two images `build-<short-sha>/provider-aws-amd64` and `build-<short-sha>/provider-aws-controller-amd64`. You can tag & push these images. But that's pretty much what you need to start using those images; you can remove the steps where you `kubectl crossplane provider build` and `kubectl crossplane provider push` because the images you'll push with `docker push` are ready to be used.

Reply:
I see, makes sense. Thanks for the clarification. For some reason I could not find the code for `kubectl crossplane build|push provider` in  the crossplane-cli repo.

Reply:
Sure thing! That repo has been archived, here is the CLI code https://github.com/crossplane/crossplane/tree/master/cmd/crank

Reply:
ah cool, thanks.